src/library/librarycontrol.cpp

```diff
@@ -279,37 +292,94 @@ void LibraryControl::slotSelectPrevTrack(double v) {
 }
 
 void LibraryControl::slotSelectTrack(double v) {
-    if (m_pLibraryWidget == NULL) {
+    if (!m_pLibraryWidget) {
         return;
     }
 
     int i = (int)v;
 
-    LibraryView* activeView = m_pLibraryWidget->getActiveView();
+    auto activeView = m_pLibraryWidget->getActiveView();
     if (!activeView) {
         return;
     }
     activeView->moveSelection(i);
 }
 
-void LibraryControl::slotSelectSidebarItem(double v) {
-    if (m_pSidebarWidget == NULL) {
-        return;
+
+
+void LibraryControl::slotMoveUp(double v) {
+    if (v > 0) {
+        emitKeyEvent(QKeyEvent{QEvent::KeyPress, Qt::Key_Up, Qt::NoModifier});
     }
+}
+
+void LibraryControl::slotMoveDown(double v) {
     if (v > 0) {
-        QApplication::postEvent(m_pSidebarWidget, new QKeyEvent(
-            QEvent::KeyPress,
-            (int)Qt::Key_Down, Qt::NoModifier, QString(), true));
-        QApplication::postEvent(m_pSidebarWidget, new QKeyEvent(
-            QEvent::KeyRelease,
-            (int)Qt::Key_Down, Qt::NoModifier, QString(), true));
-    } else if (v < 0) {
-        QApplication::postEvent(m_pSidebarWidget, new QKeyEvent(
-            QEvent::KeyPress,
-            (int)Qt::Key_Up, Qt::NoModifier, QString(), true));
-        QApplication::postEvent(m_pSidebarWidget, new QKeyEvent(
-            QEvent::KeyRelease,
-            (int)Qt::Key_Up, Qt::NoModifier, QString(), true));
+        emitKeyEvent(QKeyEvent{QEvent::KeyPress, Qt::Key_Down, Qt::NoModifier});
+    }
+}
+
+void LibraryControl::slotMoveVertical(double v) {
+    const auto key = (v < 0) ? Qt::Key_Up: Qt::Key_Down;
+    const auto times = static_cast<unsigned short>(v);
```

Comment:
Ah, no I meant for negative values of v, times will be large since it's not representable:
```static_cast<unsigned short>(-1.0) == 23232```
I think you want ```static_cast<unsigned short>(std::abs(v))```